### PR TITLE
GLIB exploration for online NSRT learning approach

### PR DESCRIPTION
**Pull request by tomsilver**

### CODE REVIEW

File: src/approaches/online_nsrt_learning_approach.py [hunk 1]
```diff
@@ -72,17 +83,85 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
...
+        Assumes that the size of the atom set is between CFG.glib_min_goal_size
+        and CFG.glib_max_goal_size (inclusive).
+        """
+        assert CFG.glib_min_goal_size <= len(atoms) <= CFG.glib_max_goal_size
+        canonical_atoms = self._get_canonical_lifted_atoms(atoms)
+        # Note minus sign: less frequent is better.
+        return -self._novelty_counts[canonical_atoms]
+
+    @staticmethod
+    def _get_canonical_lifted_atoms(
```
Comment (ronuchit) [inline src/approaches/online_nsrt_learning_approach.py:136]: does this belong in utils?

Comment (ronuchit) [reply src/approaches/online_nsrt_learning_approach.py:136]: "no" is a reasonable response

File: src/approaches/online_nsrt_learning_approach.py [hunk 2]
```diff
@@ -45,14 +52,18 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
         # explored at each timestep is a hyperparameter. The train task
         # is randomly selected.
 
-        # Create the explorer. Note that GLIB and greedy lookahead are not yet
-        # supported.
+        # Create the explorer. Note that greedy lookahead is not yet supported.
+        preds = self._get_current_predicates()
         explorer = create_explorer(CFG.explorer,
```
Comment (ronuchit) [inline src/approaches/online_nsrt_learning_approach.py:57]: so would this crash if CFG.explorer == random_options bc things like babble_predicates and atom_score_fn can't be passed as kwargs?

**Review by ronuchit — APPROVED** [review]
also update the supercloud script? (maybe that's still a draft PR)

**Comment by tomsilver** [comment]
>also update the supercloud script? (maybe that's still a draft PR)

thanks yeah will do

File: src/approaches/online_nsrt_learning_approach.py [hunk 3]
```diff
@@ -45,14 +52,18 @@ def get_interaction_requests(self) -> List[InteractionRequest]:
         # explored at each timestep is a hyperparameter. The train task
         # is randomly selected.
 
-        # Create the explorer. Note that GLIB and greedy lookahead are not yet
-        # supported.
+        # Create the explorer. Note that greedy lookahead is not yet supported.
```
[the remark this answers is not in the document]
Comment (tomsilver) [reply src/approaches/online_nsrt_learning_approach.py:55]: right

File: src/approaches/online_nsrt_learning_approach.py (hunk 2 above)
Comment (tomsilver) [reply src/approaches/online_nsrt_learning_approach.py:57]: no it won't crash -- that stuff is handled within `create_explorer`

File: src/approaches/online_nsrt_learning_approach.py (hunk 1 above)
Comment (tomsilver) [reply src/approaches/online_nsrt_learning_approach.py:136]: yeah i think no, for now at least

**Pull request merged**